pkg/minishift/registration/redhat_test.go

```diff
@@ -112,22 +121,14 @@ func TestRedHatRegistratorUnregister(t *testing.T) {
 	}
 	commander := provision.GenericSSHCommander{Driver: d}
 	registrator := NewRedHatRegistrator(commander)
-	s.CommandToOutput["sudo subscription-manager version"] = `server type: This system is currently not registered.`
-	if err := registrator.Unregister(param); err != nil {
-		t.Fatal("Distribution shouldn't able to unregister")
-	} else {
-		cmd := "sudo subscription-manager unregister"
-		if _, ok := s.Commands[cmd]; ok{
-			t.Fatalf("Expected command: %s", cmd)
-		}
-	}
+
 	s.CommandToOutput["sudo subscription-manager version"] = `server type: RedHat Subscription Management`
 	if err := registrator.Unregister(param); err != nil {
-		t.Fatal("Distribution should able to unregister")
+		t.Fatal("Distribution should be able to unregister")
 	} else {
 		cmd := "sudo subscription-manager unregister"
 		if _, ok := s.Commands[cmd]; !ok{
-			t.Fatalf("Expected command: %s", cmd)
+			t.Fatal("Expected command: sudo subscription-manager unregister")
```

Comment:
@hferentschik s.Commands contain all commands executed on the mock ssh server, during unregister what happen is it execute `sudo subscription-manager unregister` and Commands map have that entry. Now if we put `cmd` variable then error message would be misleading. I got what is confusion here so I did some more modification.